tests/class-all-test.php

```diff
@@ -813,4 +815,43 @@ public function test_get_current_url( $force_https, $url, $expected ) {
 		$res = self::$parsely->get_current_url();
 		self::assertStringStartsWith( $expected, $res );
 	}
+
+	/**
+	 * Test the wp_parsely_post_type filtter
+	 */
+	public function test_filter_wp_parsely_post_type() {
+		$options = get_option( \Parsely::OPTIONS_KEY );
+
+		$post_array = $this->create_test_post_array();
+		$post_id    = $this->factory->post->create( $post_array );
+		$post_obj   = get_post( $post_id );
+		$this->go_to( '/?p=' . $post_id );
+
+		// Try to change the post type to an allowed value - BlogPosting
+		add_filter('wp_parsely_post_type', function() {
+			return 'BlogPosting';
+		});
+
+		$metadata = self::$parsely->construct_parsely_metadata( $options, $post_obj );
+		self::assertSame( 'BlogPosting', $metadata['@type'] );
+
+		// Do not run the following assertion for PHP 5.6
+		if ( version_compare( PHP_VERSION, '7.0.0', '<' ) ) {
+			return;
+		}
+
+		// Try to change the post type to a non-allowed value - Not_Allowed
+		add_filter('wp_parsely_post_type', function() {
+			return 'Not_Allowed_Type';
+		});
+
+		/**
+		 * Ideally we use two methods expectWarning and expectWarningMessageMatches to test this error
+		 * But they're not available until PHPUnit 8.4 while here we're still running PHPUnit 7.x
+		 * @see 7.5 https://phpunit.readthedocs.io/en/7.5/writing-tests-for-phpunit.html#testing-php-errors
+		 * @see 8.4 https://phpunit.readthedocs.io/en/8.4/writing-tests-for-phpunit.html#testing-php-errors-warnings-and-notices
+		 */
+		self::expectException( PHPUnit_Warning::class );
+		self::$parsely->construct_parsely_metadata( $options, $post_obj );
```

Comment:
I just think it's important to test this case. 
I am not sure if I am making this over-complicated? Or should I just simple test against `NewsArticle` as it's the fall-back value for this case: 

https://github.com/Parsely/wp-parsely/blob/ffc769f1858e886d9aded9e1a36c691d614add61/src/class-parsely.php#L971-L972